src/core/image.js

```diff
@@ -506,6 +506,109 @@ var PDFImage = (function PDFImageClosure() {
       }
     },
 
+    /**
+     * Added by RamSoft (11/5/2015) to resize large resolution PDFS 
+     * as to improve rendering time
+     */
+    resizeGrayPixels: function(imgData, comps, bpc) {
+      var result = false;
+
+      //we only resize 1 bit per pixel or 8 bits per pixel grayscale image
+      if (comps === 1 && (bpc === 1 || bpc === 8)) {
```

Comment:
there is no need to check this here, move these checks out of the function and return new imgData instead